### ISSUE TITLE
python based ATMS SDR converter for netCDF4 and h5 files

### ISSUE THREAD

**Issue by BenjaminRuston** [JCSDA-internal/ioda-converters#721]
## Description
create a  python converter for ATMS SDR data in netCDF4 (NASA) or hdf5 (NOAA CLASS) to IODA2

## Requirements
This allows IODA2 files to be created from raw files from NASA or NOAA

## Acceptance Criteria (Definition of Done)
IODA2 files are successfully created from ATMS SDRs

## Dependencies
this converter uses the python h5py interface library

**Comment by srherbener** [comment]
@BenjaminRuston I installed h5py in my python setup on my Mac and I'm running into a library version conflict with the jedi-stack.
- h5py is hdf5 library version 1.12.1
- jedi-stack is hdf5 library version 1.12.0

The converter is creating a corrupted output file and the hdf5 version conflict might be the culprit. (It could be something else however - I haven't got to the root cause yet.)

Have you been successful at getting this converter to produce a vaild hdf5 file?

**Comment by rickgrubin** [comment]
If updating `HDF5` in the `jedi-stack` isn't an option, you may have to perform a [custom installation of h5py](https://docs.h5py.org/en/latest/build.html#custom-installation).

**Comment by srherbener** [comment]
Thanks for the pointer @rickgrubin. I'll give that a try.

**Comment by srherbener** [comment]
@rickgrubin the custom build of h5py worked. I used this mode:
```
CC="mpicc" HDF5_MPI="ON" HDF5_DIR=/path/to/parallel-hdf5 pip install --no-binary=h5py h5py
```
to get the interface to the parallel io feature of hdf5.

Thanks!

**Comment by srherbener** [comment]
@BenjaminRuston I'm getting a valid hdf5 file out of the converter now.

The output spec (-o option) of this script is a directory (the test is naming the directory atms_obs.nc4 which is confusing). The actual output ioda file is appearing in `testrun/atms_obs.nc4/atms_obs.nc4` of which the compare script doesn't know how to handle.

I'm guessing that the converter needs the output  spec to be a directory because it looks like it might eventually run parallel jobs and create multiple files. 

Is this correct? If so, I can update the compare script to handle the  intermediate directory structure.

However, if the eventual function of the script is to create multiple files and concatenate them back together into one file, then perhaps it makes more sense to change the options on the converter to specify a directory for the intermediate files and (separately) the path to the final output file. 

What do you think?

**Comment by srherbener** [comment]
I just realized I can change the -o argument to "testrun" and get the output file where it needs to be for now. I'll go ahead and do that and we can worry about this issue for later (since we are under a time crunch right now).

**Comment by BenjaminRuston** [comment]
yes it should be changed to have user input for ultimate file name output.  I'm still learning what is standard, and turns out not a lot.  The passing of the analysis time is generally done, but not universally used in the filename.  I may go ahead and change this to:

dtg_datatype_optional.nc4

dtg = 2021121500
datatype = atms
optional = sdr

but change it to pass the output filename.

**Comment by srherbener** [comment]
I've changed -o to specify the path to the output file.

The nccmp step in the test is taking 10 minutes to run since the files have 22K locations. Would it be okay for me to trim the test files down to 100 or so  locations?

**Comment by BenjaminRuston** [comment]
yes,,, I guess it's possible but this file is what you get from the server

I noticed it was very very slow... it was not on the conversion step but on the nc compare step.. this files write very quickly... can you confirm

**Comment by srherbener** [comment]
@BenjaminRuston I concur with the fast file write time and slow compare time. I figured out how to pare down the test files so I went ahead and did that. The test runs in about 1/2 second now.

I saved the original input file in the repo for now so it can be used for H(x) testing. I put the reduced test input data in a new file: `SNDR_SNPP_ATMS_SAMPLE.nc4`.

I also combined the obs_data and loc_data dictionaries. Ie, the output file now gets the metadata.

The converter appears to be working now. However, it might be missing some metadata like with the GNSSRO converter.

**Comment by srherbener** [comment]
Also, I've pushed all of these changes to github.

**Comment by BenjaminRuston** [comment]
thanks Steve, yes will take a crack at MetaData now that François and I just did this for #711
